lib/core/src/lib/datatable/data/object-datatable-adapter.ts

```diff
@@ -121,35 +121,33 @@ export class ObjectDataTableAdapter implements DataTableAdapter {
 
         if (sorting?.key) {
             this._rows.sort((a: DataRow, b: DataRow) => {
-                let left = a.getValue(sorting.key);
-                let right = b.getValue(sorting.key);
-
-                if (typeof left === 'number' && typeof right === 'number') {
-                    return sorting.direction === 'asc' ? left - right : right - left;
-                } else {
-                    if (left) {
-                        left = left instanceof Date ? left.valueOf().toString() : left.toString();
-                    } else {
-                        left = '';
-                    }
+                let left = a.getValue(sorting.key) ?? '';
+                let right = b.getValue(sorting.key) ?? '';
 
-                    if (right) {
-                        right = right instanceof Date ? right.valueOf().toString() : right.toString();
-                    } else {
-                        right = '';
-                    }
+                if (typeof left !== 'string') {
+                    left = left.valueOf().toString();
```

Comment:
it's a very risky assumption that everything that is not the string is a date... it can be a number or an object for example

Reply:
it is safe, method `valueOf` exists for every not empty data type